ios/browser/skus/skus_service_factory.cc

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/skus/skus_service_factory.h"
+
+#include "base/feature_list.h"
+#include "brave/components/skus/browser/skus_service.h"
+#include "brave/components/skus/browser/skus_utils.h"
+#include "brave/components/skus/common/features.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "components/keyed_service/ios/browser_state_dependency_manager.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "ios/chrome/browser/browser_state/browser_state_otr_helper.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/web/public/browser_state.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace skus {
+
+// static
+mojo::PendingRemote<mojom::SkusService> SkusServiceFactory::GetForBrowserState(
+    ChromeBrowserState* browser_state) {
+  // Return null if feature is disabled
+  if (!base::FeatureList::IsEnabled(skus::features::kSkusFeature)) {
```

Comment:
we should keep all the checks in `BuildServiceInstanceFor` and just check for null here from `GetServiceForBrowserState`

Reply:
Fixed in d5f872f2faa3404fa3f32295885b7682d8da890b